wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,874 @@
+        """Implements template processing mechanizm similar to `oc process`.
+        It does to functions
+          1. parametrized templates have to be processed in order to replace parameters with values.
+          2. templates consist of list of objects. Those objects have to be extracted
+          before creation accordingly.
+
+        Args:
+            name: (str) template name
+            namespace: (str) openshift namespace
+            parameters: parameters and values to replace default ones
+        Return: list of objects stored in template
+        """
+        # workaround for bug https://github.com/openshift/openshift-restclient-python/issues/60
+        raw_response = self.o_api.read_namespaced_template(name=name, namespace=namespace,
+                                                           _preload_content=False)
+        raw_data = json.loads(raw_response.data)
+        updated_data = self.rename_structure(raw_data)
+        read_template = self.ociclient.V1Template(**updated_data)
+        if parameters:
+            updated_template = self.update_template_parameters(template=read_template, **parameters)
+        else:
+            updated_template = read_template
+        raw_response = self.o_api.create_namespaced_processed_template(namespace=namespace,
+                                                                       body=updated_template,
+                                                                       _preload_content=False)
+        raw_data = json.loads(raw_response.data)
+        updated_data = self.rename_structure(raw_data)
+        processed_template = self.ociclient.V1Template(**updated_data)
+        return processed_template.objects
+
+    def rename_structure(self, struct):
+        """Fixes inconsistency in input/output data of openshift python client methods
+
+        Args:
+            struct: data to process and rename
+        Return:
+        """
+        if not isinstance(struct, (str, unicode)) and isinstance(struct, Iterable):
+            if isinstance(struct, dict):
+                for key in struct.keys():
+                    # we shouldn't rename something under data or spec
+                    if key == 'stringData':
+                        # this key has to be renamed but its contents should be left intact
+                        struct[inflection.underscore(key)] = struct.pop(key)
+                    elif key in ('spec', 'data', 'string_data'):
+                        # these keys and data should be left intact
+                        pass
+                    else:
+                        # all this data should be processed and updated
+                        val = self.rename_structure(struct.pop(key))
+                        struct[inflection.underscore(key)] = val
+                return struct
+            else:
+                for item in struct:
+                    self.rename_structure(item)
```

Comment:
I would personally not do this as it is changing the iterable on the fly. This would be a better alternative:

```python
struct = [self.rename_structure(item) for item in struct]
```